### ISSUE TITLE
bug: Appium can't install WDA with Xcode 15.0.1 - xcodebuild failed with code 70

### ISSUE THREAD

**Issue by fleytman** [appium/appium#19386]
### Do I have the most recent component updates?

- [X] I use the most recent available driver/plugin and server versions

### Is the component officially supported by the Appium team?

- [X] I have verified the component repository is present under the Appium organization in GitHub

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

After upgrade Xcode to 17.0.**1** appium can't install WDA. Error `xcodebuild failed with code 70`

### Expected Behavior

Appium can install WDA

### Minimal Reproducible Example

{
  "platformName": "iOS",
  "appium:platformVersion": "16.4",
  "appium:deviceName": "iPhone 13",
  "appium:automationName": "XCUITest"
}

### Environment

- Operating system: macos  Sonoma 14.1.1
- Xcode without problem 15.0
- Xcode **with** problem 15.0.**1**
- Appium server version (output of `appium --version`): 2.2.1
- Appium driver(s) and their version(s): xcuitest@5.8.2 [installed (npm)]
- Appium plugin(s) and their version(s):
- Node.js version (output of `node --version`): v21.1.0
- `npm` version (output of `npm --version`): 10.2.0
- Platform and version under test: ios 16.4
- Real device or emulator/simulator: simulator

### Link to Appium Logs

https://gist.github.com/fleytman/1a9006e6ffe4a90e1924e35cd2496a8e

### Further Information

At the moment, tracks with similar problems have been closed, with myself and a few of my coworkers experiencing the problem. To solve it, I tried

Upgrade macos
Update appium and drivers
Build WDA manually (error)
Remove DerivedData
Remove simulators
Re-download simulators

**Comment by KazuCocoa** [comment]
```
[XCUITestDriver@2613 (de1cbb33)] Beginning test with command 'xcodebuild build-for-testing test-without-building -project /Users/userName/.appium/node_modules/appium-xcuitest-driver/node_modules/appium-webdriveragent/WebDriverAgent.xcodeproj -scheme WebDriverAgentRunner -derivedDataPath /Users/userName/Library/Developer/Xcode/DerivedData/WebDriverAgent-fswxndkorumgkhfrutcrmcawyttn -destination id=954E35C3-4E39-4350-8EF1-54597AD3D4E1 IPHONEOS_DEPLOYMENT_TARGET=16.4 GCC_TREAT_WARNINGS_AS_ERRORS=0 COMPILER_INDEX_STORE_ENABLE=NO' in directory '/Users/userName/.appium/node_modules/appium-xcuitest-driver/node_modules/appium-webdriveragent'
[XCUITestDriver@2613 (de1cbb33)] Output from xcodebuild will only be logged if any errors are present there. To change this, use 'showXcodeLog' desired capability
[XCUITestDriver@2613 (de1cbb33)] Waiting up to 60000ms for WebDriverAgent to start
[XCUITestDriver@2613 (de1cbb33)] Matched '/status' to command name 'getStatus'
[XCUITestDriver@2613 (de1cbb33)] Proxying [GET /status] to [GET http://127.0.0.1:8100/status] with no body
[XCUITestDriver@2613 (de1cbb33)] connect ECONNREFUSED 127.0.0.1:8100
[XCUITestDriver@2613 (de1cbb33)] Matched '/status' to command name 'getStatus'
[XCUITestDriver@2613 (de1cbb33)] Proxying [GET /status] to [GET http://127.0.0.1:8100/status] with no body
[XCUITestDriver@2613 (de1cbb33)] connect ECONNREFUSED 127.0.0.1:8100
[Xcode] xcodebuild exited with code '70' and signal 'null'
```


What happens when you run the below command directly?

```
xcodebuild build-for-testing test-without-building -project /Users/userName/.appium/node_modules/appium-xcuitest-driver/node_modules/appium-webdriveragent/WebDriverAgent.xcodeproj -scheme WebDriverAgentRunner -derivedDataPath /Users/userName/Library/Developer/Xcode/DerivedData/WebDriverAgent-fswxndkorumgkhfrutcrmcawyttn -destination id=954E35C3-4E39-4350-8EF1-54597AD3D4E1 IPHONEOS_DEPLOYMENT_TARGET=16.4 GCC_TREAT_WARNINGS_AS_ERRORS=0 COMPILER_INDEX_STORE_ENABLE=NO
```

**Comment by sonumuto** [comment]
I am also having the same issue. When I run the command above it works just fine. But, I have to run the command manually which is not ideal for automation.

**Comment by KazuCocoa** [comment]
`appium:showXcodeLog` capability helps to check the xcodebuild log details in Appium, could you share the full appium log with the capability and the manual command itself?

As Appium, it only can adjust `xcodebuild` argument. I wondered if both had the same `derivedDataPath`, or not when different result occurred against the same project file.


So far, I observed the 70 error when the xcodebuild itself failed to find the target device's udid. It was xcodebuild internal thing though

**Comment by fleytman** [comment]
@KazuCocoa, thank you for your help. I was able to figure out the problem. There were two reasons:

1. In my attempts to solve the issue, I deleted all simulator images, including iOS 17. Xcodebuild was complaining about it, even though I was trying to launch a session on the 16th simulator.
2. Updating XCUITest should be done with the following command: `appium driver update xcuitest --unsafe`. Without the `--unsafe` flag, it wasn't updating.

Now the issue is resolved for me and one of my colleagues. I'm conducting a survey to see if anyone else is still experiencing the problem.
